Signal/src/call/CallAudioService.swift

```diff
@@ -169,32 +176,52 @@ struct AudioSource: Hashable {
             return
         }
 
+        // Disallow bluetooth when user has explicitly chosen the built in receiver.
+        // I'm actually not sure why this is required - it seems like we should just be able
+        // to setPreferredInput to call.audioSource.portDescription in this case,
+        // but in practice I'm seeing the call revert to the bluetooth headset.
+        // Presumably something else (in WebRTC?) is touching our shared AudioSession.
+        let options: AVAudioSessionCategoryOptions = call.audioSource?.isBuiltInEarPiece == true ? [] : [.allowBluetooth]
```

Comment:
This is pretty ham fisted, but it works. If you have a bluetooth in your ear, but for some reason, have opted to not use it, you may get a little blip of ringtone as the call connects.